ci/jenkins/pipelines/skuba-upgrade-from-4.2.Jenkinsfile

```diff
@@ -0,0 +1,133 @@
+    
+    environment {
+        SKUBA_BINPATH = "/home/jenkins/go/bin/skuba"
+        OPENSTACK_OPENRC = credentials('ecp-openrc')
+        GITHUB_TOKEN = credentials('github-token')
+        VMWARE_ENV_FILE = credentials('vmware-env')
+        TERRAFORM_STACK_NAME = "${BUILD_NUMBER}-${JOB_NAME.replaceAll("/","-")}".take(70)
+     }
+
+     stages {
+
+        stage("Build skuba"){
+            steps{
+                sh(script: "git checkout ${INITIAL_VERSION}")
+                sh(script: "make -f Makefile install", label: "Build Skuba ${INITIAL_VERSION}")
+            }
+        }
+
+        stage('Provision cluster') {
+            steps {
+                sh(script: 'make -f ci/Makefile provision', label: 'Provision')
+            }
+        }
+
+        stage('Deploy cluster') {
+            steps {
+                sh(script: 'make -f ci/Makefile deploy KUBERNETES_VERSION=${KUBERNETES_VERSION}', label: 'Deploy')
+                sh(script: 'make -f ci/Makefile check_cluster', label: 'Check cluster')
+            }
+        }
+
+        stage('upgrade admin node') {
+            environment {
+                REG_CODE = credentials('scc-regcode-caasp')
+            }
+            steps {
+                echo 'upgrading sles'
+                sh "sudo sudo zypper in -y --no-recommends zypper-migration-plugin"
+                sh "sudo SUSEConnect -r ${REG_CODE}"
+                sh "sudo zypper migration --migration 2 --no-recommends --non-interactive --auto-agree-with-licenses"
+                echo "upgrade completed"
+            }
+        }
+    
+        stage('reboot'){
+            agent { label 'caasp-team-private-integration' } 
+            environment {
+                JENKINS_ID = credentials('jenkins-ssh')
+            }
+            steps{
+                echo 'rebooting ${WORKER_HOST} ...'
+                sh "ssh -i ${JENKINS_ID} -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -T jenkins@${WORKER_HOST} sudo reboot &"
+               
+                sleep 60
+ 
+                timeout(180){
+                    waitUntil { script {
+                        sh(script: "ssh -i ${JENKINS_ID} -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -T jenkins@${WORKER_HOST} true", returnStatus: true) == 0
+                    }}
+                }
+            }
+        }
+        
+        stage('upgrade skuba') { 
+            steps {
+                sh(script: "sudo zypper in -y --no-recommends go1.13")
+                sh(script: "git checkout -f ${TARGET_VERSION}")
+                sh(script: "make install", label: "Build ${TARGET_VERSION}")
+            }   
+        }
+
+        stage('upgrade cluster'){
+            environment {
+                REG_CODE = credentials('scc-regcode-caasp')
+                
+            }
+            steps {
+                sh(script: "make -f ci/Makefile test SUITE='test_upgrade_from_4_2' SKIP_SETUP='deployed'", label: "Test upgrade")
```

Comment:
should we make the test suite a param of the job?

Reply:
We could, however, this job is specific for that test so I don't see any advantage. Moreover, I see some potential issues because the test is the actual upgrade process. Please, clarify which use cases you see it would be necessary to have this as a parameter.

Reply:
yeah, I was thinking when moving this into a shared library, having the test suite as a parameter so we could reuse it.

Butr it doesnt make much sense in this case, or now.